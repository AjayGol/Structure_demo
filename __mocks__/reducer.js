import { combineReducers } from 'redux';
import AccountReducer from '../src/modules/account/reducer';
import BrokerReducer from '../src/modules/broker/reducer';
import HomeReducer from '../src/modules/home/reducer';
import SearchReducer from '../src/modules/search/reducer';
import WatchlistReducer from '../src/modules/watchlist/reducer';
import OrdersReducer from '../src/modules/watchlist/reducer';
import NotificationReducer from '../src/modules/notification/reducer';
import buyAssetReducer from '../src/modules/buy-assets/reducer';

export default combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});
