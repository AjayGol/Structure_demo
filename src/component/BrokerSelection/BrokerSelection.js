import React, { useState, useEffect } from 'react';
import { StyleSheet, View, TouchableOpacity, Image } from 'react-native';
import constant from '../../helper/constant';
import Animated, { Easing } from 'react-native-reanimated';
import { UserEvent } from '../../helper/fabricHelper/track';

const CustomAlert = props => {
  const [selectedCell, setSelectedCell] = useState(-1);

  const {
    container,
    selectedCellContainer,
    selectedCellSubContainer,
    imageSelected,
    fullContainer,
    imageOneSelected,
    lineSeparate,
  } = styles;

  const { right, isIpad, listingData } = props.data;

  const onSelectType = type => {
    setSelectedCell(type);
    props.onAlertClick(right, type, selectedCell);
  };

  return (
    <View style={[container, constant.isIpad && isIpad]}>
      {listingData && (
        <View style={selectedCellContainer}>
          {listingData &&
            listingData.map((obj, i) => {
              return (
                <TouchableOpacity
                  onPress={() => onSelectType(i)}
                  key={i}
                  style={selectedCellSubContainer}>
                  {obj.image && (
                    <Image
                      resizeMode="contain"
                      style={
                        (selectedCell === i && imageOneSelected) ||
                        imageSelected
                      }
                      source={{
                        uri: obj.image,
                      }}
                    />
                  )}
                  <View style={fullContainer} />
                  {i !== listingData.length - 1 && (
                    <View style={lineSeparate} />
                  )}
                </TouchableOpacity>
              );
            })}
        </View>
      )}
    </View>
  );
};

const BrokerSelectionPopUp = props => {
  const [fadeAnim] = useState(new Animated.Value(0));

  useEffect(() => {
    setTimeout(() => {
      Animated.timing(fadeAnim, {
        duration: 200,
        toValue: 1,
        easing: Easing.out(Easing.linear),
      }).start();
    }, 5);
  });

  const onAlertClick = (title, typeSelected = false) => {
    hideAlert(title, typeSelected);
  };

  const hideAlert = (title, typeSelected) => {
    const { type } = props.data;
    setTimeout(() => {
      Animated.timing(fadeAnim, {
        toValue: 0,
        duration: 100,
        easing: Easing.out(Easing.linear),
      }).start(() => {
        props.onAlertClick(title, type, typeSelected);
      });
    }, 50);
  };

  const onPressClose = () => {
    if (props.onAlertClick) {
      props.onAlertClick();
    }
  };

  const { closeScreen } = styles;

  return (
    <Animated.View
      style={[
        styles.alertOuter,
        { opacity: fadeAnim },
        !!constant.isiPAD && { alignItems: 'center' },
      ]}>
      <TouchableOpacity onPress={onPressClose} style={closeScreen} />
      <CustomAlert data={props.data} onAlertClick={onAlertClick} />
    </Animated.View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 6,
    backgroundColor: '#FFFFFF',
    marginHorizontal: 23,
    elevation: 22,
  },
  alertOuter: {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    position: 'absolute',
    paddingHorizontal: 11,
    backgroundColor: 'rgba(5,34,61,0.4)',
    justifyContent: 'center',
    zIndex: 111111,
    elevation: 2,
  },
  isIpad: {
    maxWidth: 400,
    width: '100%',
  },
  selectedCellContainer: {
    backgroundColor: '#FFFFFF',
    borderRadius: 4,
    borderColor: 'rgba(89, 89, 92, 0.07)',
    borderWidth: 1,
    paddingVertical: 5,
    paddingHorizontal: 17,
  },
  selectedCellSubContainer: {
    paddingVertical: 10,
    flexDirection: 'row',
    alignItems: 'center',
  },
  imageSelected: { width: 93, height: 20, tintColor: constant.linearGrayColor },
  imageOneSelected: {
    width: 93,
    height: 20,
    tintColor: constant.yelloMainColor,
  },
  fullContainer: { flex: 1 },
  lineSeparate: {
    position: 'absolute',
    left: -17,
    right: -17,
    bottom: 0,
    borderColor: 'rgba(89, 89, 92, 0.07)',
    borderWidth: 1,
    height: 1,
  },
  closeScreen: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
  },
});

export default BrokerSelectionPopUp;
