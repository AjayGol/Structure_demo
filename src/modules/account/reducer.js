import {
  SET_SAFE_AREA_INTENT,
  API_WAITING_TIME,
  START_LOADING,
  SET_USER_DETAIL,
  CUSTOM_POPUP,
  EVENT_DATA,
  SET_USER_OTHER_DETAIL,
} from './constant';

export const initStateAccount = {
  email: process.env.NODE_ENV === 'development' ? 'ketan@gmail.com' : '',
  password: process.env.NODE_ENV === 'development' ? 'test1234' : '',
  userName: '',
  token: '',
  safeAreaInsetsData: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDataChat: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDefault: { top: 0, bottom: 0, left: 0, right: 0 },
  apiWaitingTime: 0,
  isLoading: false,
  customPopUp: { isShow: false, data: {} },
  userDetails: {
    id: '',
    name: '',
    username: '',
    mobile: '',
    password: '',
    birthday: '',
    gender: '',
    address: [],
    tradingFrequency: 1,
    brokers: [],
    brokerLoginDetails: {},
    tradingFamiliarity: [],
    traderProgress: {
      traderReturns: 0,
      followerReturns: 0,
      AUM: 0,
      followedPortfolioCount: 0,
      _id: '',
      trader: '',
      id: '',
      twitterUserData: {
        isLoggedIn: false,
      },
    },
    userDetailsOtherData: {
      watchlistsLength: 0,
      ifFirstTradeExecuted: false,
    },
    profileCompletionRatio: {
      profile: 0,
      twitter: 0,
      pan: 0,
      brokers: 0,
      knowledge: 0,
    },
    eventData: false,
  },
};

export default (state = initStateAccount, action) => {
  switch (action.type) {
    case SET_SAFE_AREA_INTENT: {
      return {
        ...state,
        safeAreaInsetsDefault: action.payload,
        safeAreaInsetsData: action.payload,
      };
    }
    case START_LOADING: {
      return {
        ...state,
        isLoading: action.payload,
      };
    }
    case API_WAITING_TIME: {
      return {
        ...state,
        apiWaitingTime: action.payload,
      };
    }
    case SET_USER_DETAIL: {
      return {
        ...state,
        userDetails: action.payload,
        userDetailsOtherData: action.payloadAll,
      };
    }
    case SET_USER_OTHER_DETAIL: {
      return {
        ...state,
        userDetailsOtherData: action.payload,
      };
    }
    case CUSTOM_POPUP: {
      return {
        ...state,
        customPopUp: action.payload,
      };
    }
    case EVENT_DATA: {
      return {
        ...state,
        eventData: action.payload,
      };
    }

    default:
      return state;
  }
};
