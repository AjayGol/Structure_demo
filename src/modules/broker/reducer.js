import { ANGEL_URL } from './constant';

export const initStateBroker = {
  angelURL: '',
};

export default (state = initStateBroker, action) => {
  switch (action.type) {
    case ANGEL_URL: {
      return {
        ...state,
        angelURL: action.payload,
      };
    }

    default:
      return state;
  }
};
