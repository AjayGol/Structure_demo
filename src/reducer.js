import { combineReducers } from 'redux';
import AccountReducer from './modules/account/reducer';
import HomeReducer from './modules/home/reducer';
import SearchReducer from './modules/search/reducer';
import BrokerReducer from './modules/broker/reducer';
import NotificationReducer from './modules/notification/reducer';
import { initStateAccount } from './modules/account/reducer';
import { initStateHome } from './modules/home/reducer';
import WatchlistReducer from './modules/watchlist/reducer';
import { initStateWatchlist } from './modules/watchlist/reducer';
import { initStateOrders } from './modules/orders/reducer';
import OrdersReducer from './modules/orders/reducer';
import { initStateSearch } from './modules/search/reducer';
import { initStateBroker } from './modules/broker/reducer';
import { initStateNotification } from './modules/notification/reducer';
import { initStateBuyAsset } from './modules/buy-assets/reducer';
import buyAssetReducer from './modules/buy-assets/reducer';

export const appDefaultReducer = {
  account: initStateAccount,
  home: initStateHome,
  orders: initStateOrders,
  search: initStateSearch,
  broker: initStateBroker,
  watchlist: initStateWatchlist,
  buyAsset: initStateBuyAsset,
  notification: initStateNotification,
};

const appReducer = combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});

export default function rootReducer(state, action) {
  let finalState = appReducer(state, action);
  if (action.type === 'RESET_STORE') {
    finalState = appDefaultReducer; //resetReducer(finalState, action);
  }
  return finalState;
}
