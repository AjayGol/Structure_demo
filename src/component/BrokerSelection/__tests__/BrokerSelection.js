import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import reducer from '../../../../__mocks__/reducer';
import { Provider } from 'react-redux';
import BrokerSelection from '../BrokerSelection';
const store = createStore(reducer);

let tree;
let title = 'Testing';
beforeEach(() => {
  tree = shallow(
    <Provider store={store}>
      <BrokerSelection />
    </Provider>,
  );
});

test(`${title} - renders correctly`, () => {
  expect(tree).toMatchSnapshot();
});
