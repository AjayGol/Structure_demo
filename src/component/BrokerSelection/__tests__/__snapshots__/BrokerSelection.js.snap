// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - renders correctly 1`] = `ShallowWrapper {}`;
